### PR TITLE
Promisify and support Node.js >=4

### PR DESCRIPTION
**Pull request by kevva**
Plus some small tweaks :).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
+		const handlers = data && data[0].LSHandlers;
 
 		if (!handlers || handlers.length === 0) {
-			return cb(null, bundleId);
+			return bundleId;
 		}
 
-		for (var i = 0; i < handlers.length; i++) {
-			var el = handlers[i];
+		for (let i = 0; i < handlers.length; i++) {
```
Comment (SamVerschueren) [inline index.js]: Is for..of available in node 4?

File: test.js [hunk 2]
```diff
@@ -1,11 +1,6 @@
...
+import test from 'ava';
+import m from './';
 
-it('should return the default browser', function (cb) {
-	defaultBrowser(function (err, id) {
-		console.log('Bundle id:', id);
-		assert(/^com\./.test(id));
-		cb();
-	});
+test('return the default browser', async t => {
```
Comment (SamVerschueren) [inline test.js]: remove title if it's only one test

File: index.js (hunk 1 above)
Comment (kevva) [inline index.js]: Yes.

File: cli.js [hunk 3]
```diff
@@ -1,21 +1,12 @@
...
 
-defaultBrowserId(function (err, id) {
-	if (err) {
-		console.error(err.message);
-		process.exit(1);
-	}
-
-	console.log(id);
-});
+defaultBrowserId().then(id => console.log(id));
```
Comment (sindresorhus) [inline cli.js]:
``` js
defaultBrowserId().then(console.log);
```

File: index.js [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
-var file = untildify(osxVersion >= 14 ? '~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist' : '~/Library/Preferences/com.apple.LaunchServices.plist');
+const os = require('os');
+const bplist = require('bplist-parser');
+const untildify = require('untildify');
+const pify = require('pify');
+const osxVersion = Number(os.release().split('.')[0]);
+const file = untildify(osxVersion >= 14 ? '~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist' : '~/Library/Preferences/com.apple.LaunchServices.plist');
 
-module.exports = function (cb) {
+let bundleId = 'com.apple.Safari';
```
Comment (sindresorhus) [inline index.js]: Define this inside the function.

File: index.js [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
+
+module.exports = () => {
 	if (process.platform !== 'darwin') {
-		throw new Error('Only OS X systems are supported');
+		return Promise.reject(new Error('Only OS X systems are supported'));
```
Comment (sindresorhus) [inline index.js]: > Only OS X is supported

File: .travis.yml [hunk 6]
```diff
@@ -1,5 +1,8 @@
 sudo: false
 language: objective-c
+before_install:
+  - brew update
+  - brew unlink node
 install: brew install node
 script:
   - npm install
```
Comment (sindresorhus) [inline .travis.yml]: You can test on OS X with travis now:

``` yml
os: osx
language: node_js
node_js:
  - '5'
  - '4'
```

File: index.js [hunk 7]
```diff
@@ -1,36 +1,34 @@
...
 
 		if (!handlers || handlers.length === 0) {
-			return cb(null, bundleId);
+			return bundleId;
 		}
 
-		for (var i = 0; i < handlers.length; i++) {
-			var el = handlers[i];
+		for (const el of handlers) {
+			const el = handlers[i];
```
Comment (sindresorhus) [inline index.js]: Remove :arrow_up:

File: .travis.yml (hunk 6 above)
Comment (kevva) [inline .travis.yml]: Ohh, that's nice!

File: index.js (hunk 7 above)
Comment (kevva) [inline index.js]: Oops x_x.

**Comment by sindresorhus** [comment]
👍🍻

**Comment by kevva** [comment]
:beers:

**Pull request merged**